### ISSUE TITLE
personal.unlockAccount(foo) errors

### ISSUE THREAD

**Issue by q9f** [etclabscore/core-geth#104]
```
> personal.unlockAccount()
GoError: usage: unlockAccount(account, [ password, duration ])
        at native
        at <eval>:1:1(2)

> personal.unlockAccount(eth.coinbase)
GoError: usage: unlockAccount(account, [ password, duration ])
        at native
        at <eval>:1:24(4)
```

It should prompt for a password and not error. Latest master @ a31ca3b55ecfb09a67e0baf060c7d16901d182e0

**Comment by meowsbits** [comment]
I am unable to reproduce this.

```js
> admin.nodeInfo.name
"CoreGeth/v1.11.5-unstable-a31ca3b5-20200520/linux-amd64/go1.13.5"
> eth.accounts
["0x7d38c531e7544625a4ea96437d7fe5c2918f3307"]
> eth.coinbase
"0x7d38c531e7544625a4ea96437d7fe5c2918f3307"
> personal.unlockAccount(eth.accounts[0]);
Unlock account 0x7d38c531e7544625a4ea96437d7fe5c2918f3307
Passphrase:
true
> personal.unlockAccount(eth.coinbase);
Unlock account 0x7d38c531e7544625a4ea96437d7fe5c2918f3307
Passphrase: 
true

```

Can you confirm that `eth.coinbase` is indeed set and available in your `--keystore`?

**Comment by mr-wiggie** [comment]
I had this same error (and found your question while searching for a solution). I can replicate.

```
> personal.unlockAccount(eth.accounts[0])

GoError: usage: unlockAccount(account, [ password, duration ])
        at native
        at <eval>:1:24(6)
```

When I include password it is successful, just the password prompt is failing.

```
> personal.unlockAccount(eth.accounts[0], "my_password", 100)

true
```



I agree that `personal.unlockAccount(account)` should work, and prompt for a password if required.

Also coinbase is set and in keystore.

**Comment by q9f** [comment]
Ok, I think I am able to reproduce this by attaching a foundation geth instance to a core geth instance. I'm not sure this is a valid issue then. Still weird.

**Comment by meowsbits** [comment]
I reproduced... but I'm not sure what I did differently :facepalm: :question: 

I've experimented with `attach`, `--datadir`, and `--keystore` permutations and I can't find a pattern.
Will look at this more in-depth today.

Note that the example below is using ethereum/go-ethereum at v1.9.14.

```sh
> ./build/bin.foundation/geth --nodiscover console 2> /dev/null
Welcome to the Geth JavaScript console!

instance: Geth/v1.9.14-stable-6d74d1e5/linux-amd64/go1.13.5
coinbase: 0xce05110d3ab580cb94a32ce3db049d32edbc49df
at block: 0 (Wed Dec 31 1969 18:00:00 GMT-0600 (CST))
 datadir: /home/ia/.ethereum
 modules: admin:1.0 debug:1.0 eth:1.0 ethash:1.0 miner:1.0 net:1.0 personal:1.0 rpc:1.0 txpool:1.0 web3:1.0

> personal.listAccounts
["0xce05110d3ab580cb94a32ce3db049d32edbc49df", "0xd4584b5f6229b7be90727b0fc8c6b91bb427821f", "0xb01801cbfb695d14d445fe9ba9bc2d65681d3f06", "0xd4584b5f6229b7be90727b0fc8c6b91bb427821f", "0x7b9759ba0da6b628d0fad656cf25fd95ba944a62", "0xd4584b5f6229b7be90727b0fc8c6b91bb427821f", "0x3d239bf7c402ea1bf2f8d07507c731457ec9a483"]
> var myAccount = "0x3d239bf7c402ea1bf2f8d07507c731457ec9a483";
undefined
> personal.unlockAccount(myAccount);
GoError: usage: unlockAccount(account, [ password, duration ])
        at native
        at <eval>:1:24(3)

> personal.unlockAccount(myAccount, "");
true
```

**Comment by meowsbits** [comment]
This is the issue. https://github.com/ethereum/go-ethereum/pull/21081/files

It was not included in `v1.9.14` (see `3666da8`):
```sh
| | | * 3666da8a4b E Felix Lange console: fix unlockAccount argument count check (#21081) (13 days ago) 
| | | * f3f1e59eea E Marius van der Wijden accounts/abi: simplify reflection logic (#21058) (2 weeks ago) 
| | | * 677724af0c E Satpal cmd: fix log contexts (#21077) (2 weeks ago) 
| | | * 46698d7931 E Péter Szilágyi params: begin v1.9.15 release cycle (2 weeks ago) 
| | | * 6d74d1e5f7 E Péter Szilágyi (HEAD, tag: v1.9.14, foundation/release/1.9) params: release go-ethereum v1.9.14 (2 weeks ago) 
```

**Comment by meowsbits** [comment]
This fix is already included in core-geth v1.11.4 (and on master). 

That's why, @q9f, you were able to reproduce it by attaching the foundation instance (which must have been at their v1.9.14 tag, or before the fix which is now on their master, but hasn't been included in one of their tagged release yet).

@mr-wiggie Can you confirm this by checking and sharing the geth version you were using?

```sh
$ geth version
```

or 

```js
> admin.nodeInfo.name
```

I expect you were on ethereum/go-ethereum before the merged fix linked above.

---

Below demonstrates again that the issue is already resolved with core-geth:
```sh
> ./build/bin/geth --datadir ./temp-datadir --nodiscover console 2>/dev/null
Welcome to the Geth JavaScript console!

instance: CoreGeth/v1.11.4-stable-478a00cd/linux-amd64/go1.13.5
at block: 0 (Wed Dec 31 1969 18:00:00 GMT-0600 (CST))
 datadir: /home/ia/go/src/github.com/ethereum/go-ethereum/temp-datadir
 modules: admin:1.0 debug:1.0 eth:1.0 ethash:1.0 miner:1.0 net:1.0 personal:1.0 rpc:1.0 txpool:1.0 web3:1.0

> personal.listAccounts
[]
> personal.newAccount();
Passphrase:
Repeat passphrase:
"0xbe12059484daccff1065f7027c2f7cb150aaf334"
> personal.listAccounts
["0xbe12059484daccff1065f7027c2f7cb150aaf334"]
> personal.unlockAccount(personal.listAccounts[0]);
Unlock account 0xbe12059484daccff1065f7027c2f7cb150aaf334
Passphrase:
true
```

**Comment by meowsbits** [comment]
Closing as
- [x] a go-ethereum issue, and
- [x] resolved
